src/infra/core/src/sync_service_impl.rs

```diff
@@ -493,6 +506,23 @@ impl SyncServiceImpl {
         }
     }
 
+    fn new_watermark(block: &MetadataBlock) -> Option<DateTime<Utc>> {
+        match &block.event {
+            MetadataEvent::AddData(add_data) => {
+                if add_data.new_watermark.is_some() {
+                    return add_data.new_watermark;
+                }
+            }
+            MetadataEvent::ExecuteTransform(transform) => {
+                if transform.new_watermark.is_some() {
+                    return transform.new_watermark;
```

Comment:
See description of the field: https://github.com/open-data-fabric/open-data-fabric/blob/master/schemas/metadata-events/AddData.json#L30

If `new_watermark` is present it doesn't necessarily mean that it was updated by the last operation. Once watermark appears it is "carried over" from one block to another. This is done so that when constructing next operation we didn't have to scan too deep to understand what the current watermark is.

Reply:
Then why it is called so confusing? Name it `last_watermark` instead.
So, I guess I basically have to go through the range and detect changes in this field on the updated interval.

Reply:
If follows the `prev*` / `new*` convention for input/output fields of some operation introduced as part of the [RFC-014](https://github.com/open-data-fabric/open-data-fabric/blob/master/rfcs/014-minimize-offset-scanning.md). I can see how it can be confusing in this case, but also how `last_watermark` can be confusing for a transfrom - e.g. is it a watermark before or after the transformation?

Reply:
Reverted